BoxContentSDK/BoxContentSDK/Requests/BOXFileRequest.m

```diff
@@ -50,9 +50,16 @@ - (BOXAPIOperation *)createOperation
     NSDictionary *queryParameters = nil;
 
     if (self.requestAllFileFields) {
-        queryParameters = @{BOXAPIParameterKeyFields: [self fullFileFieldsParameterString]};
-    }
+        NSString *fieldString = [self fullFileFieldsParameterString];
+        
+        // Include the representations field in the request supporting the request for file representation
+        if ([self.httpHeaderValueXRepHints length] > 0) {
```

Comment:
Given the implementation above, this makes sense to me. (I'm surprised that there isn't an `isEmpty` function that could be used instead, perhaps? But that's a different thing.)

The idea here is that `fields=representations` should only be requested if `X-Rep-Hints` will be provided. How the API is structured to provide the content of `X-Rep-Hints` is debatable, but in the end, it comes down to a string used as a header and if it's empty, we don't want to ask for the `representations` field.